components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```diff
@@ -0,0 +1,487 @@
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement createSubscriberPrepStmt = new NamedPreparedStatement(connection,
+                CREATE_SUBSCRIBER_SQL);
+             NamedPreparedStatement createSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
+                     CREATE_SUBSCRIBER_PROPERTIES_SQL)) {
+
+            createSubscriberPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+            createSubscriberPrepStmt.setString(ENTITLEMENT_MODULE_NAME, holder.getModuleName());
+            createSubscriberPrepStmt.setInt(TENANT_ID, tenantId);
+            createSubscriberPrepStmt.executeUpdate();
+
+            for (PublisherPropertyDTO dto : holder.getPropertyDTOs()) {
+                if (dto.getId() != null && StringUtils.isNotBlank(dto.getValue())) {
+
+                    createSubscriberPropertiesPrepStmt.setString(PROPERTY_ID, dto.getId());
+                    createSubscriberPropertiesPrepStmt.setString(DISPLAY_NAME, dto.getDisplayName());
+                    createSubscriberPropertiesPrepStmt.setString(PROPERTY_VALUE, dto.getValue());
+                    createSubscriberPropertiesPrepStmt.setBoolean(IS_REQUIRED, dto.isRequired());
+                    createSubscriberPropertiesPrepStmt.setInt(DISPLAY_ORDER, dto.getDisplayOrder());
+                    createSubscriberPropertiesPrepStmt.setBoolean(IS_SECRET, dto.isSecret());
+                    createSubscriberPropertiesPrepStmt.setString(MODULE, dto.getModule());
+                    createSubscriberPropertiesPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                    createSubscriberPropertiesPrepStmt.setInt(TENANT_ID, tenantId);
+
+                    createSubscriberPropertiesPrepStmt.addBatch();
+                }
+            }
+            createSubscriberPropertiesPrepStmt.executeBatch();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            IdentityDatabaseUtil.rollbackTransaction(connection);
+            throw new EntitlementException("Error while inserting subscriber details", e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(connection);
+        }
+    }
+
+    /**
+     * DAO method to update a subscriber.
+     *
+     * @param subscriberId        subscriber ID.
+     * @param updatedModuleName   updated module name.
+     * @param updatedPropertyDTOS updated property DTOs.
+     * @param tenantId            tenant ID.
+     * @throws EntitlementException If an error occurs.
+     */
+    private void updateSubscriber(String subscriberId, String updatedModuleName,
+                                  PublisherPropertyDTO[] updatedPropertyDTOS, int tenantId)
+            throws EntitlementException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement updateSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
```

Comment:
Shall we move this inside the,
`if (ArrayUtils.isNotEmpty(updatedPropertyDTOS)) {`